Source/Orts.Simulation/Simulation/Signalling/Signals.cs

```diff
@@ -3869,7 +3870,7 @@ public void BreakDownRouteList(Train.TCSubpathRoute reqRoute, int firstRouteInde
         public List<int> ScanRoute(Train thisTrain, int firstSectionIndex, float firstOffset, int firstDirection, bool forward,
                 float routeLength, bool honourManualSwitch, bool autoAlign, bool stopAtFacingSignal, bool reservedOnly, bool returnSections,
                 bool searchFacingSignal, bool searchBackwardSignal, bool searchFacingSpeedpost, bool searchBackwardSpeedpost,
-                bool isFreight, bool considerSpeedReset = false)
+                bool isFreight, bool considerSpeedReset = false, bool checkReenterOriginalRoute = false)
```

Comment:
this is an awful list of parameters and should be candidate for refactoring (separately)